src/test/java/com/hotels/beeju/HiveServer2CoreTest.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (C) 2015-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.beeju;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+import java.io.IOException;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.junit.Test;
+
+import com.hotels.beeju.core.BeejuCore;
+import com.hotels.beeju.core.HiveServer2Core;
+
+public class HiveServer2CoreTest {
+
+  private final BeejuCore core = new BeejuCore();
+  private final HiveServer2Core hiveServer2Core = new HiveServer2Core(core);
+
+  @Test
+  public void initiateServer() throws InterruptedException {
+    hiveServer2Core.before();
+    assertThat(hiveServer2Core.getJdbcConnectionUrl(),
+        is("jdbc:hive2://localhost:" + hiveServer2Core.getPort() + "/" + core.databaseName()));
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void closeServer() {
+    hiveServer2Core.after();
```

Comment:
similar here, you're not testing what you want to test. You get a nullpointer exception because `hiveServer2Core..before()` is not called which means `hiveServer2Core.getHiveServer2()` returns null which means `NPE`.  The `.getServiceState()` can tell you if it is started or stopped.

Reply:
ah yes i see, i'll change these tests